client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx

```diff
@@ -74,25 +75,39 @@ const googleWorkspaceCardInformation: ProviderCard = {
 };
 
 const GoogleWorkspaceCard = ( {
+	cartDomainName,
 	comparisonContext,
-	currencyCode = '',
 	detailsExpanded,
-	domain,
-	gSuiteProductMonthly,
-	gSuiteProductYearly,
-	hasCartDomain,
 	intervalLength,
 	onExpandedChange,
 	selectedDomainName,
-	selectedSite,
-	shoppingCartManager,
 	source,
 }: EmailProvidersStackedCardProps ): ReactElement => {
 	const googleWorkspace: ProviderCard = { ...googleWorkspaceCardInformation };
 	googleWorkspace.detailsExpanded = detailsExpanded;
 
-	const gSuiteProduct =
-		intervalLength === IntervalLength.MONTHLY ? gSuiteProductMonthly : gSuiteProductYearly;
+	const hasCartDomain = Boolean( cartDomainName );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const selectedSite = useSelector( getSelectedSite );
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSite?.ID ) );
+	const domain = getSelectedDomain( {
+		domains,
+		selectedDomainName: selectedDomainName,
+	} ) as unknown;
+
+	const cartKey = useCartKey();
+	const shoppingCartManager = useShoppingCart( cartKey );
+
+	const gSuiteProductYearly = useSelector( ( state ) =>
+		getProductBySlug( state, GOOGLE_WORKSPACE_BUSINESS_STARTER_YEARLY )
+	);
+	const gSuiteProductMonthly = useSelector( ( state ) =>
+		getProductBySlug( state, GOOGLE_WORKSPACE_BUSINESS_STARTER_MONTHLY )
+	);
+
+	const gSuiteProduct: ProductListItem = ( intervalLength === IntervalLength.MONTHLY
+		? gSuiteProductMonthly
+		: gSuiteProductYearly ) as ProductListItem;
```

Comment:
Do we really need `as ProductListItem` here? Can we not type the two values returned from the `useSelector` call? (We may need null/undefined checks to be safe - I don't see those in this new code.)